### PR TITLE
Update section_controller and other ruby code to prep for assignment dropdown updates

### PR DESCRIPTION
**Pull request by dmcavoy**
Breaking apart https://github.com/code-dot-org/code-dot-org/pull/45042/files

- Passes in the locale code to the new summary of course offering and course versions
- Updates the section_controller.rb to use a before action to get unit and course from the params.
- Updates the section_controller to only accept `script_id` instead of allowing for `script_id` or `script` as params. I searched and this was the only path still use the `script` param to create or update.

## Links

[Jira](https://codedotorg.atlassian.net/browse/PLAT-1584)
[Eng Plan - Self Paced Pl](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#heading=h.d0b9vmlcrsox)
[Mini Spec - Course Assignment Dropdown Updates](https://docs.google.com/document/d/11-dtCFuSmST_vP2MdkH0H0lb2e5AGFu1MxiWSRsnkns/edit#)

## Testing story

- Add/updated unit tests

### CODE REVIEW

File: dashboard/app/controllers/api/v1/sections_controller.rb [hunk 1]
```diff
@@ -59,9 +56,7 @@ def create
     render head :bad_request unless section
 
     # TODO: Move to an after_create step on Section model when old API is fully deprecated
-    if script_to_assign
-      current_user.assign_script script_to_assign
-    end
+    current_user.assign_script @unit if @unit
```
Comment (davidsbailey) [inline dashboard/app/controllers/api/v1/sections_controller.rb:59]: is the comment still valid? which old API are we talking about?

File: dashboard/test/controllers/api/v1/sections_controller_test.rb [hunk 2]
```diff
@@ -523,11 +519,7 @@ class Api::V1::SectionsControllerTest < ActionController::TestCase
       login_type: Section::LOGIN_TYPE_EMAIL,
       course_id: pilot_unit_group.id
     }
-    assert_response :success
-    # TODO: Better to fail here?
-
-    assert_nil returned_json['course_id']
-    assert_nil returned_section.unit_group
+    assert_response :bad_request
```
Comment (davidsbailey) [inline dashboard/test/controllers/api/v1/sections_controller_test.rb]: I think `:forbidden` would be more appropriate here

**Review by davidsbailey — APPROVED** [review]

File: dashboard/app/controllers/api/v1/sections_controller.rb (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/controllers/api/v1/sections_controller.rb:59]: We have two sections_controllers. One in /api/vi and one that is not. I did not pull on the threads of why we have two but it was not necessary to update the other controller to make things work here.

Comment (dmcavoy) [reply dashboard/app/controllers/api/v1/sections_controller.rb:59]: I asked about it in this slack thread: https://codedotorg.slack.com/archives/C0T10HG6N/p1646679987426589

**Pull request merged**